### PR TITLE
update connector specs and definitions with new .com docs urls

### LINKED ISSUE #17568: Update documentationUrl to new domain

### PR DESCRIPTION
**Pull request by pedroslopez**
## What 

Connectors are providing documentationUrls that are pointing to `docs.airbyte.io` instead of the new(er) `docs.airbyte.com`. This is taking users to outdated documentation.

EDIT: The DNS issue that was preventing the redirect from occurring is now fixed so this is lower priority, but I still think these should be updated.

closes #17568 

## How
- Update documentationUrl in source_definitions.yaml and destination_definitions.yaml
- Update documentationUrl in connector's specs (spec.json, spec.yaml)
- Update generators to point to docs.airbyte.com instead of docs.airbyte.io in specs when bootstrapping a new connector

### CODE REVIEW

**Comment by pedroslopez** [comment]
Running tests for strict encrypt connectors since they have `expected_spec.json`s

File: airbyte-integrations/connectors/destination-postgres-strict-encrypt/src/test/resources/expected_spec.json [hunk 1]
```diff
@@ -143,7 +143,7 @@
               },
               "client_key_password": {
                 "type": "string",
-                "title": "Client key password (Optional)",
```
Comment (pedroslopez) [inline airbyte-integrations/connectors/destination-postgres-strict-encrypt/src/test/resources/expected_spec.json:146]: These weren't updated when we removed the optionals so the spec test was failing.

**Comment by pedroslopez** [comment]
Failing tests for connectors are unrelated and known to be failing. I ensured that the GetSpec test in each was still passing - most are failing on Discover, probably hinting at missing credentials or something.

**Comment by pedroslopez** [comment]
I removed some of the docs changes that wre originally in this PR (changing URLs in connector READMEs to try to make this a bit more reviewable, though there's still quite a bit of files touched. Hopefully it's not too bad to review since it should all be .io -> .com changes.)

**Review by alafanechere — APPROVED** [review]
LGTM,
I think we'd ideally want spec update to lead to a new connector version. 
I'm wondering if we could have taken a different lazy approach:
1. add an acceptance test that checks if `documentationUrl` point to `https://docs.airbyte.io`
2. consider that the fix will eventually be provided by connector maintainers

**Review by tuliren — APPROVED** [review]
:shipit:

**Pull request merged**